fix and improve content on reference pages of table elements

glmvc opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

More detailed description of the current issues/inconsistencies: https://github.com/mdn/content/issues/26088

- Fix/remove/rewrite some out-of-place paragraphs
- move examples from the [`<tr>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr) reference page to the [`<table>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) reference page (like it is the case for the other table elements like `<td>` or `<th>`)
- improve minor things like using macros for html elements for example

### Motivation

As a reader, it was confusing to me how the examples on the  [`<tr>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr) reference page are directly connected to the <tr> element itself. They are in fact general table examples...
There were also paragraphs that fit better on the overall  [`<table>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) reference page, e.g. mentioning the table section in the learning area.

### Additional details

I have been waiting for feedback on the corrections/improvements in the [related issue](https://github.com/mdn/content/issues/26088) for quite a while now: But I thought I will just start making the improvements and fixes and create a pr.

Personally, I think the other table element reference pages are fine now. However, I somehow have the feeling that there is too much information on the [`<table>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) reference page now. Maybe that was the reason why the examples where on the  [`<tr>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr) page? But as already stated, they were too general there.

Perhaps some duplicate/redundant information can be removed. Further suggestions or improvements are welcome.

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/26088


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

glmvc committed:
fix and improve content on reference pages of table elements

estelle committed:
basic table

Changed the first table to use best practices.
changed "simple" to "basic"

estelle commented on files/en-us/web/html/element/table/index.md:
````diff
@@ -269,15 +285,449 @@ td {
+</table>
+```
+
+The differences that matter here—for the purposes of discussing row and column spans—are in the first few lines of the code above. Note the use of `rowspan` to make the "Name", "ID", and "Balance" headers occupy two rows instead of just one, and the use of `colspan` to make the "Membership Dates" header cell span across two columns.
````
```suggestion
Because the dates area spans two rows, the other headers also need to span two rows. We include the `rowspan` attribute on the other  {{HTMLElement("th")}} elements  to make the "Name", "ID", and "Balance" headers occupy two rows. We use the `colspan` attribute to make the "Membership Dates" header cell span the two date columns.
```

estelle replied:
These example were indeed on the wrong page. However, they should really be on the `th` and `td` page, as these attributes are supported there, or a tutorial or how to, not the table page or the tr page

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -269,15 +285,449 @@ td {
+
+The differences that matter here—for the purposes of discussing row and column spans—are in the first few lines of the code above. Note the use of `rowspan` to make the "Name", "ID", and "Balance" headers occupy two rows instead of just one, and the use of `colspan` to make the "Membership Dates" header cell span across two columns.
+
+The CSS is unchanged from before.
+
```
```suggestion
```
I don't think we need to mention this since we are saying we're continuing the previous examples.

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -142,29 +146,41 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
+      <td>Johnson</td>
+      <td>2014</td>
+    </tr>
+  </tbody>
```
The original example didn't show best practices. We also want to avoid "simple. And we should iterated that tables are for tabular data

estelle reviewed (commented):
Examples deep diving into an attribute should be on the element's page that support that attribute. Colspan and Rowspan should be on td and th, but as they're not....

The four basic examples can be improved, and then linked back to their relevant page. For example, the `<tfoot>` example should contain some footer content, like "total". the `<col style>` should be a class and the  `<col span="2">` should also have a class to show that something is happening. All "simple" words should be removed, and all tables should have a `thead`, `body`, and `caption`.

As all the table content pages lead here (i don't think they should), each example should be separate and link back to the elements / attribute that is being highlighted. 



I started providing some feedback on content. I hope that is helpful. I also merged in a "fix" to the first example which had issues. You definitely identified something that needs to be fixed!

glmvc committed:
Apply suggestions from code review

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc commented:
@estelle thank you very much for your feedback and suggestions, I appreciate it!
As mentioned above, I had been waiting for feedback in the [related issue](https://github.com/mdn/content/issues/26088) for some time, but unfortunately had not received any response. So I thought I'd start and make some changes.

The best would probably be to have a matching and representative example on each of the table-related element pages, as is actually the case on the[ `<tbody>` reference page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody#examples).

This would allow us to make the [`<table>` reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) less cluttered by using only **one overall best practice example** and, as you mentioned, linking the details to the appropriate reference pages.

Now, I don't know if it makes sense that I start working on it... After all, it is a somewhat larger task and restructuring (expanding the table-related element pages with suitable examples and explanations; uncluttering the `<table>` page itself and leaving just one overall example). What do you think?
Should I just start to the best of my knowledge and judgement or do we approach it in a more structured way with concrete suggestions?

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -269,15 +285,448 @@ td {
+
+##### HTML
+
+The HTML for the data within the table is similar to the previous example, except for the addition of the new column in each data row. The main changes are in the header. The creation of the super-heading involves adding row and column spans to `<th>` elements to merge elements from multiple rows and columns.  The header spans two rows to create heading cells that are correctly lined up with the  ...
```
```suggestion
The HTML for the data within the table is similar to the previous example, except for the addition of the new column in each data row. The main changes are in the header. The creation of the super-heading involves adding row and column spans to `<th>` elements to merge elements from multiple rows and columns. The header spans two rows to create heading cells that are correctly lined up with the content in the body of the table.
```

estelle committed:
Update files/en-us/web/html/element/table/index.md

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -269,15 +285,448 @@ td {
+
+##### Result
+
+The output is entirely unchanged, despite the addition of useful contextual information under the hood:
```
```suggestion
Visually, the output is unchanged.  The additions provide useful contextual information for assistive technologies, including screen readers and search engines:
```

estelle commented:
i commented on https://github.com/mdn/content/issues/26088#issuecomment-1641078125

glmvc commented:
All right, thanks - I'm on it :)

glmvc committed:
Update files/en-us/web/html/element/table/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Apply suggestions from code review

reported by reviewdog 🐶

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc committed:
Merge branch 'main' into fixing-table-content

estelle reviewed (commented):
I reviewed this a month ago, but forgot to hit "summit review". Apologies.

glmvc committed:
Apply suggestions from code review

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Merge branch 'main' into fixing-table-content

glmvc commented:
@estelle No problem. I haven't really had time to work on it yet. In the next few days, I'll have more time to make progress. But it will still take quite a bit of work to completely finish this task.

Here is some feedback on some of your suggestions after some work:
- > All "simple" words should be removed
  
  In my opinion, it is fine to use the term "simple" to indicate a very basic table example without elements like `tbody` or `thead`. This brings me to the next point...
- > and all tables should have a thead, body, and caption.

  I understand that the use of these elements should reflect good practice. But creating a complete table structure for every simple example would, IMHO, only detract from explaining the use of the actual element being referenced. As you surely know, table structures can easily result in lots of lines.... And AFAIK, it's completely fine to have a table with only `tr` elements, as the other elements are optional. (Please correct me if I'm interpreting the spec wrong).

- > As all the table content pages lead here (i don't think they should)

  As I said, I will try to declutter the table reference page and add, move or create examples to the corresponding element reference pages.

- > Advanced styling should be on a styling tables guide page
  > Table sorting should also be on its own tutorial page

  I will try to avoid excessive styling examples, as there is already a [detailed article](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Styling_tables) on this in the CSS Learning Area.
  Some sections might do well on the table reference page after decluttering. A whole tutorial page would probably be overkill for most parts, I guess. But let's review this at the end.


I totally agree with the rest of your suggestions. Thank you for your help so far :)

estelle commented:
> * > All "simple" words should be removed
>   
>   
>   In my opinion, it is fine to use the term "simple" to indicate a very basic table example without elements like `tbody` or `thead`. This brings me to the next point...

We don't use "easy" or "simple" -- because it isn't for some people --  but the word "basic" is fine.

glmvc committed:
fix, adapt, and unify table elements

glmvc committed:
Merge branch 'main' into fixing-table-content

glmvc commented:
@estelle, Sorry for the delay. I was quite busy, and it was a bit of work.

I made some decisions while improving and unifying the content. Sometimes I wasn't quite sure if they were the right decisions, but I think it's a good improvement overall:
- unify the overall structure, sections, etc.
- correct, complete and add missing information (e.g., attributes)
- add missing note boxes for deprecated attributes - maybe too much under every attribute?
- add missing "Usage notes"
- update the "See also" sections
- introduce examples on every table-related page
- fix, improve and adapt interactive examples; note https://github.com/mdn/interactive-examples/pull/2667, which should ideally be merged at the same time
- create illustrations for abstract concepts (column and row spanning) - I also put the HTML markup for the screenshots in the directories, but I don't know if this is wanted and permitted?
- try to declutter the overall table reference page by linking to the corresponding pages

There is still room for improvement, in my opinion, but I think it's now far easier to make adaptations as the structure of the pages is more identical.

I'm also not quite sure about my writing style, since I tried to keep a neutral tone. The "Practical table example" on the table page had a more "user-interactive" voice. I'm not sure about this example at all... I didn't want to change it too much because I want some feedback first to see if my changes are going in the right direction.

I hope that's all for now. If something is not clear, or you require some changes to be made, feel free to ask anytime :)

estelle commented on files/en-us/web/html/element/col/index.md:
````diff
@@ -26,52 +24,161 @@ The following attributes are deprecated and should not be used. They are documen
+  </tr>
+</table>
+```
+
+```css
````
```suggestion

### CSS

We use CSS instead of deprecated HTML attributes to provide a background color to the columns and align the cell content:
```css
table {
  border-collapse: collapse;
}
caption {
  caption-side: bottom;
}
th, td {
  border: 1px solid rgb(140 140 140);
  text-align: center;
}
```

estelle replied:
it's important to show the text-align being on the td and th.

estelle commented on files/en-us/web/html/element/col/index.md:
````diff
@@ -26,52 +24,161 @@ The following attributes are deprecated and should not be used. They are documen
+```
+
+```css hidden
+table {
+  border-collapse: collapse;
+  border: 2px solid rgb(140, 140, 140);
+  font-family: sans-serif;
+  font-size: 0.8rem;
+  letter-spacing: 1px;
+}
+
+caption {
+  caption-side: bottom;
+  padding: 10px;
+}
+
+td,
+th {
+  border: 1px solid rgb(160, 160, 160);
+  padding: 8px 6px;
+}
+
+td {
+  text-align: center;
+}
````
```suggestion
table {
  font-family: sans-serif;
  font-size: 0.8rem;
  letter-spacing: 1px;
}
```

estelle replied:
moved the "you should see this" stuff to be visible

estelle commented on files/en-us/web/html/element/col/index.md:
```diff
@@ -26,52 +24,161 @@ The following attributes are deprecated and should not be used. They are documen
+
+## Usage notes
+
+- The `<col>` element may be used within a {{HTMLElement("colgroup")}} element that doesn't have a `span` attribute.
+- It's important to note that `<col>` elements do not structurally group columns together, as this is the role of the {{HTMLElement("colgroup")}} element.
```
```suggestion
The `<col>` element is used within a {{HTMLElement("colgroup")}} element that doesn't have a `span` attribute. The `<col>` elements does not structurally group columns together. This is the role of the {{HTMLElement("colgroup")}} element.

Only a limited number of CSS properties affect columns:
-   {{cssxref("border")}}: The various `border` properties apply, but only if the `<table>` has {{cssxref("border-collapse", "border-collapse: collapse")}} set. 
- {{cssxref("background")}} : The various`background` properties will set the background for cells within the column, but will only be visible if the cell and row backgrounds are transparent. 
- {{cssxref("width")}}: The `width` property defines a minimum width for the column, as if {{cssxref("min-width")}} were set. 
- {{cssxref("visibility")}}: The {{cssxref("visibility")}} property is also supported.

```

estelle reviewed (commented):
Thanks. You added a lot of useful and necessary information. I only had time to do  partial review; most of it making the content more MDN-esque. I'll continue on Monday.

glmvc commented:
Thank you @estelle. I hope it is not that bad and requires a lot of work for you...
I will review your suggestions when you are through.

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -7,378 +7,669 @@ browser-compat: html.elements.table
-### Simple table
+You'll find some examples below, but for more examples and an in-depth tutorial, see the [HTML tables](/en-US/docs/Learn/HTML/Tables) series in our [Learn web development](/en-US/docs/Learn) area, where you'll learn how to use the table elements and their attributes to get just the right structuring for your tabular data. Visit the [Styling tables](/en-US/docs/Learn/CSS/Building_blocks/Styling_ta ...
+
+Since the structure of a `<table>` involves the use of several table-related HTML elements along with various associated attributes, the following examples are intended to provide a simplified explanation that covers the basics and common standards. Additional and more detailed information can be found on the corresponding linked pages.
```
explaining that each all the examples are building on each other, with examples being an h2, each individual example within the larger example being an h3, the HTML, CSS headings being h4, and no h5s.

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -7,378 +7,669 @@ browser-compat: html.elements.table
-### Further simple examples
+### Practical table structure, accessibility and styling
+
+This practical table example will guide you through the process of creating a complete and {{Glossary("accessibility", "accessible")}} table structure in HTML and styling it with [CSS](/en-US/docs/Web/CSS) using common practices.
+
+The goal of this more detailed example of a basic table is to create an accessible yet visually appealing table that displays a list of people's names along with information about their membership in a club.
+
+#### Basic table setup
+
+Because of the way a `<table>` must be structured in HTML, the {{Glossary("markup", "markup")}} can quickly grow into numerous lines of code. For this reason, it's important to have a clear idea of what the table should look like in the end. If the HTML markup of the table is prepared in a good and logical way, it's not only {{Glossary("semantics", "semantically")}} correct and helpful for assist ...
+
+##### HTML
+
+First, the very basic HTML structure of the table is created. There are no table content groups such as a head or foot, no cells span multiple rows or columns, no relationships between cells are explicitly defined, and no table caption is provided—for now. Basic styling is then added to create lines around the components of the table to make the data structure clearer.
+
+The table comprises four rows ({{HTMLElement("tr")}} elements), including a header row ({{HTMLElement("th")}} elements within the first row). Each row contains four columns, including a header column ({{HTMLElement("th")}} elements as the first child elements of the rows) and three data columns ({{HTMLElement("td")}} elements). As table sectioning elements are not used, the browser automatically  ...
```
some of the content was moved to the intro. This should just be the stuff for the 2nd example.

estelle commented on files/en-us/web/html/element/table/index.md:
````diff
@@ -7,378 +7,669 @@ browser-compat: html.elements.table
+}
+```
+
+##### Result
````
change all the h5 to h4, and change the embededliveexample links to the renamed example titles.

estelle reviewed (commented):
got a bit further. If you want to start reworking it, please do. A lot of the suggestions apple to multiple examples or multiple pages.

glmvc commented on files/en-us/web/html/element/caption/index.md:
```diff
@@ -21,67 +21,85 @@ The following attributes are deprecated and should not be used. They are documen
 - `align` {{deprecated_inline}}
 
-  - : This [enumerated](/en-US/docs/Glossary/Enumerated) attribute indicates how the caption must be aligned with respect to the table. It may have one of the following values:
+  - : This {{Glossary("enumerated", "enumerated")}} attribute indicates how the caption must be aligned with respect to the table. Possible values are:
```
I like `accepts` the most, because the attribute _still_ allows the values
```suggestion
  - : This deprecated {{Glossary("enumerated", "enumerated")}} attribute accepts `left`, `top`, `right`, or `bottom` as a value to indicate which side of the table the caption should be displayed. Use the {{cssxref("caption-side")}} and {{cssxref("text-align")}} CSS properties instead.
```

glmvc replied:
After giving it some thought, I have come up with the following sentence structure, which I think is clear and at the same time short and concise:
> This {{Glossary("enumerated", "enumerated")}} attribute defines which side of the table the caption should be displayed. Possible values are `left`, `top`, `right`, or `bottom`. Use the {{cssxref("caption-side")}} and {{cssxref("text-align")}} CSS properties instead, as this attribute is deprecated.

I'll go with this structure for the other deprecated attributes too. I hope that this is fine for you.

glmvc commented on files/en-us/web/html/element/caption/index.md:
```diff
@@ -21,67 +21,85 @@ The following attributes are deprecated and should not be used. They are documen
-      - : The caption is displayed to the right of the table.
-    - `bottom`
-      - : The caption is displayed below the table.
+    - `left`: the caption is displayed to the left of the table;
+    - `top`: the caption is displayed above the table;
+    - `right`: the caption is displayed to the right of the table;
+    - `bottom`: the caption is displayed below the table.
```
Yes, I think the notes are repetitive (mentioning the deprecation) and are too present / take up too much vertical space on the page... Especially when there are a lot of deprecated attributes.

glmvc commented on files/en-us/web/html/element/caption/index.md:
```diff
@@ -21,67 +21,85 @@ The following attributes are deprecated and should not be used. They are documen
+}
+
+th {
+  background-color: rgb(230, 230, 230);
```
You're right... Out of sheer outdated habit...

glmvc commented on files/en-us/web/html/element/caption/index.md:
```diff
@@ -21,67 +21,85 @@ The following attributes are deprecated and should not be used. They are documen
 
-If used, the `<caption>` element must be the first child of its parent {{htmlelement("table")}} element.
+- If used, the `<caption>` element must be the first child of its parent {{htmlelement("table")}} element.
+- When the {{HTMLElement("table")}} element that contains the `<caption>` is the only descendant of a {{HTMLElement("figure")}} element, you should use the {{HTMLElement("figcaption")}} element instead of `<caption>`.
```
I'll go with this small tweak, I hope your fine with it:
```suggestion
- When a `<table>` is nested within a {{HTMLElement("figure")}} as the figure's only content, it should be captioned via a {{HTMLElement("figcaption")}} for the `<figure>` instead of as a `<caption>` nested within the `<table>`.
```

glmvc commented on files/en-us/web/html/element/col/index.md:
```diff
@@ -7,12 +7,10 @@ browser-compat: html.elements.col
 {{HTMLSidebar}}
 
-The **`<col>`** [HTML](/en-US/docs/Web/HTML) element defines a column within a table and is used for defining common semantics on all common cells. It is generally found within a {{HTMLElement("colgroup")}} element.
+The **`<col>`** [HTML](/en-US/docs/Web/HTML) element defines one or more columns within a table and is used for defining common semantics among table columns. It may be used within a {{HTMLElement("colgroup")}} element that has no [`span`](/en-US/docs/Web/HTML/Element/colgroup#span) attribute defined.
```
The first one reads clearer for me.

glmvc committed:
Apply suggestions from code review

to be able to tweak the suggestions and apply them to the other table-related elements

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Apply suggestions from code review 2.0

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Merge branch 'main' into fixing-table-content

glmvc committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc committed:
Update files/en-us/web/html/element/col/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc committed:
Update files/en-us/web/html/element/colgroup/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc committed:
Apply linter suggestions

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc commented on files/en-us/web/html/element/colgroup/index.md:
```diff
@@ -27,50 +27,155 @@ The following attributes are deprecated and should not be used. They are documen
+## Usage notes
+
+- The `<colgroup>` is placed after any {{HTMLElement("caption")}} element (if used), but before any {{HTMLElement("thead")}}, {{HTMLElement("tbody")}}, {{HTMLElement("tfoot")}}, and {{HTMLElement("tr")}} elements.
+- Specifying column groups provides useful {{Glossary("semantics", "semantic")}} information and can be used when rendering for either screen or print.
```
You're right. "semantic" is not the right wording here.
However, it provides useful information, that can be used for e.g., styling, but also for `scope`. But that is mentioned on the corresponding table cell pages. So I will remove this bullet point.

glmvc commented on files/en-us/web/html/element/index.md:
```diff
@@ -170,18 +170,18 @@ These elements let you provide indications that specific parts of the text have
+| Element                     | Description                                                                                                                                                                                                                                               |
+| --------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| {{HTMLElement("caption")}}  | Specifies the caption (or title) of a table.                                                                                                                                                                                                              |
+| {{HTMLElement("col")}}      | Defines one or more columns within a table and is used for defining common semantics among table columns. It may be used within a {{HTMLElement("colgroup")}} element that has no [`span`](/en-US/docs/Web/HTML/Element/colgroup#span) attribute defined. |
```
I think this is confusing. What do you mean by that? Or is this an error?
The info, that the `colgroup` is put after any caption element belongs to the description of the `colgroup` in my opinion...
However, I've removed the wrong semantics information and adapted the sentence (similar to the wording in the latest spec).

glmvc commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -7,378 +7,669 @@ browser-compat: html.elements.table
-  - : This [enumerated](/en-US/docs/Glossary/Enumerated) attribute indicates how the table must be aligned inside the containing document. It may have the following values:
+  - : This {{Glossary("enumerated", "enumerated")}} attribute indicates how the table must be aligned inside the containing document. Possible values are:
 
     - `left`: the table is displayed on the left side of the document;
```
Oh, you're right of course ;) The wording in the old spec says "with respect to the document", which I understand how it's meant now...

glmvc committed:
apply suggested changes and adjustments

glmvc commented:
Thank you for your help and suggestions, @estelle.
I have revised, adapted and improved the documents again in response to your feedback and requests.
Well, I hope I've understood everything correctly and haven't forgotten anything...

But in my opinion, it doesn't look too bad anymore, and I think it's much better than it was.
Maybe the writing style can be adapted to fit more the style of mdn, but I'll let you decide that as a more experienced writer ;)
Let me know if you need anything else.

estelle commented on files/en-us/web/html/element/tr/index.md:
````diff
@@ -102,6 +81,10 @@ The CSS {{cssxref(":nth-of-type")}} pseudo-class is used to select every `odd` r
+The CSS {{cssxref(":nth-of-type")}} pseudo-class is used to select every `odd` row and set the {{cssxref("background-color")}} of those rows to a slightly darker tone, creating a so-called "zebra stripe" effect. This alternating background makes the rows of data in the table easier to parse and read—imagine having many rows and columns and trying to find some data in a particular row. In addition ...
+
 ```css
 tr:nth-of-type(odd) td {
````
```suggestion
tr:nth-of-type(odd) {
```
let's color the row, not the cells in the row.

glmvc replied:
you're right, it's about rows, so yes :)

estelle commented on files/en-us/web/html/element/tr/index.md:
````diff
@@ -168,6 +155,10 @@ The CSS is nearly unchanged from the [previous example](#basic_row_setup), excep
+The CSS is nearly unchanged from the [previous example](#basic_row_setup), except for some additional styling to highlight the "header row" so that the headers of the columns stand out from the other cells.
+
 ```css
 tr:nth-of-type(odd) td {
````
```suggestion
tr:nth-of-type(odd) {
```

estelle commented on files/en-us/web/html/element/tr/index.md:
```diff
@@ -134,9 +117,13 @@ td {
+An additional table row (`<tr>` element) is added as the first row of the table with column header cells ({{HTMLElement("th")}} elements). This makes it easier to understand the information in the columns and identify the data. The [`scope`](/en-US/docs/Web/HTML/Element/th#scope) attribute is used for the headers ({{HTMLElement("th")}} elements) within the first row to specify which cells they re ...
 
-The CSS is nearly unchanged from the [previous example](#basic_row_setup), except for some additional styling to highlight the "header row" so that the headings of the columns stand out from the other cells.
+> **Note:** Normally, {{HTMLElement("thead")}} and {{HTMLElement("tbody")}} elements are used to group such rows into the respective table head and body sections. This is not implemented in this example to focus on the rows and keep the example simple.
```
Let's add a thead and tbody to this example just so we have somewhat different examples that people can learn from on different pages.

estelle commented on files/en-us/web/html/element/tr/index.md:
```diff
@@ -337,8 +342,6 @@ th {
 
 {{EmbedLiveSample('Sorting_rows_with_a_click_on_header_cells', '650', '100')}}
 
```
```suggestion

> **Note:** To be usable and accessible, the header cell of each sortable column must be identifiable as a sorting button and each must define whether the column is currently sorted in ascending or descending order visually and with the [`aria-sort`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort) attribute.  See the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/)'s [sortable table example](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/) for more information.
```

glmvc replied:
Very good information on that as the example is not accessible. Thanks!

estelle reviewed (commented):
We're almost there. Most of the 60 comments (I know that seems like a lot) are just minor grammar tweaks. There are a handful are full paragraph edits or additions that need to be double checked and just a couple of questions.

estelle commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -312,28 +292,32 @@ The creation of the second header row involves adding [`colspan`](/en-US/docs/We
-It's a common and advisable practice to provide a [caption](#captions) ({{HTMLElement("caption")}} element) for the table's content, allowing users to quickly determine the table's relevance.
+#### HTML
+
+A table summary is added by using a table [caption](#captions) ({{HTMLElement("caption")}} element) as the first child element of the `<table>`.
```
it's been merged

glmvc commented on files/en-us/web/html/element/colgroup/index.md:
```diff
@@ -17,87 +17,68 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
+    > **Note:** Setting `vertical-align` on the `<colgroup>` element has no effect as {{HTMLElement("td")}} and {{HTMLElement("th")}} elements are not descendants of the `<colgroup>` element, and therefore they do not inherit from it.
     >
-    > If the table doesn't use a [`colspan`](/en-US/docs/Web/HTML/Element/td#colspan) attribute, use the `td:nth-of-type(an+b)` CSS selector per column, where `a` is the total number of the columns in the table and `b` is the ordinal position of the column in the table, e.g. `td:nth-of-type(7n+2) { vertical-align: middle; }` to center the second column cells vertically.
+    > If the table does not use a [`colspan`](/en-US/docs/Web/HTML/Element/td#colspan) attribute, use the `td:nth-of-type(an+b)` CSS selector per column, where `a` is the total number of the columns in the table and `b` is the ordinal position of the column in the table, e.g. `td:nth-of-type(7n+2) { vertical-align: middle; }` to center the second column cells vertically.
```
The sentence says that the formula is `an+b` and explains it like that.
Do we have to rewrite the whole paragraph then?

estelle replied:
i updated the rest of the paragraph. The original 7n+2 would work, but would confuse most people, so suggesting just linking to the selector page.

glmvc commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -21,85 +21,61 @@ The following attributes are deprecated and should not be used. They are documen
 Since the structure of a `<table>` involves the use of several table-related HTML elements along with various associated attributes, the following examples are intended to provide a simplified explanation that covers the basics and common standards. Additional and more detailed information can be found on the corresponding linked pages.
 
-### Basic table rows and columns
+These table examples demonstrate how a complete and {{Glossary("accessibility", "accessible")}} table is structured with HTML and styled with [CSS](/en-US/docs/Web/CSS).
```
```suggestion
These table examples demonstrate how to create an {{Glossary("accessibility", "accessible")}} table that is structured with HTML and styled with [CSS](/en-US/docs/Web/CSS).
```

glmvc commented on files/en-us/web/html/element/table/index.md:
```diff
@@ -120,25 +96,17 @@ The first row contains the header cells ({{HTMLElement("th")}} elements) that se
+#### HTML
 
-The table comprises four rows ({{HTMLElement("tr")}} elements), including a header row ({{HTMLElement("th")}} elements within the first row). Each row contains four columns, including a header column ({{HTMLElement("th")}} elements as the first child elements of the rows) and three data columns ({{HTMLElement("td")}} elements). As table sectioning elements are not used, the browser automatically  ...
+The table comprises four rows ({{HTMLElement("tr")}} elements) now, including a header row ({{HTMLElement("th")}} elements within the first row). Each row contains four columns, including a header column ({{HTMLElement("th")}} elements as the first child elements of the rows) and three data columns ({{HTMLElement("td")}} elements). As table sectioning elements are not used, the browser automatica ...
```
```suggestion
The table comprises four rows ({{HTMLElement("tr")}} elements) now, with four columns each. The first row is a row of header cells (The first row contains only {{HTMLElement("th")}} elements). Subsequent rows include a header column ({{HTMLElement("th")}} elements as the first child elements of each row) and three data columns ({{HTMLElement("td")}} elements). As table sectioning elements are not used, the browser automatically defines the content group structure, i.e., all rows are wrapped within the body of the table of an implicit {{HTMLElement("tbody")}} element.
```

glmvc replied:
corrected "column" to "columns" in the first sentence

glmvc commented on files/en-us/web/html/element/td/index.md:
```diff
@@ -16,88 +16,55 @@ The **`<td>`** [HTML](/en-US/docs/Web/HTML) element defines a cell of a table th
-    If this attribute is not set, the `left` value is assumed.
-
-    > **Note:** Do not use this attribute as it is deprecated (and not supported) in the latest standard: use the CSS {{cssxref("text-align")}} property instead.
+  - : Specifies the horizontal alignment of the data cell. The possible {{Glossary("enumerated")}} values are `left`, `center`, `right`, `justify`, and `char`. When supported, the `char` value aligns the textual content on the character defined in the [`char`](#char) attribute and on offset defined by the [`charoff`](#charoff) attribute. Use the {{cssxref("text-align")}} CSS property instead, as  ...
```
"an" was used in a suggestion before, so I will go with "an offset" instead of "the offset"

```suggestion
  - : Specifies the horizontal alignment of the data cell. The possible {{Glossary("enumerated")}} values are `left`, `center`, `right`, `justify`, and `char`. When supported, the `char` value aligns the textual content on the character defined in the [`char`](#char) attribute and an offset defined by the [`charoff`](#charoff) attribute. Use the {{cssxref("text-align")}} CSS property instead, as this attribute is deprecated.
```

glmvc commented on files/en-us/web/html/element/colgroup/index.md:
```diff
@@ -17,87 +17,68 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
 - `align` {{deprecated_inline}}
 
-  - : This {{Glossary("enumerated", "enumerated")}} attribute specifies how horizontal alignment of each column group cell content will be handled. Possible values are:
+  - : Specifies the horizontal alignment of each column group cell. The possible {{Glossary("enumerated")}} values are `left`, `center`, `right`, `justify`, and `char`. When supported, the `char` value aligns the textual content on the character defined in the [`char`](#char) attribute and on offset defined by the [`charoff`](#charoff) attribute. Note that the descendant {{HTMLElement("col")}} el ...
```
"the offset" is used instead of "an offset" in other suggestions and I think "the" is better?
```suggestion
  - : Specifies the horizontal alignment of each column group cell. The possible {{Glossary("enumerated")}} values are `left`, `center`, `right`, `justify`, and `char`. When supported, the `char` value aligns the textual content on the character defined in the [`char`](#char) attribute and the offset defined by the [`charoff`](#charoff) attribute. Note that the descendant {{HTMLElement("col")}} elements may override this value using their own [`align`](/en-US/docs/Web/HTML/Element/col#align) attribute. Use the {{cssxref("text-align")}} CSS property on the {{htmlelement("td")}} and {{htmlelement("th")}} elements instead, as this attribute is deprecated.
```

glmvc commented on files/en-us/web/html/element/th/index.md:
```diff
@@ -16,16 +16,16 @@ The **`<th>`** [HTML](/en-US/docs/Web/HTML) element defines a cell as the header
 - `scope`
 
-  - : This {{Glossary("enumerated", "enumerated")}} attribute defines the cells that the header (defined in the `<th>`) element relates to. Possible values are:
+  - : Defines the cells that the header (defined in the `<th>`) element relates to. Possible {{Glossary("enumerated", "enumerated")}} values are:
```
Is this sentence also true for the `rowgroup` and `colgroup` values or is it just referring to `row` and `col`?

estelle replied:
oh, i might have made a mistake here. Let me check and get back to you.

glmvc commented on files/en-us/web/html/element/th/index.md:
```diff
@@ -269,15 +252,15 @@ tr:nth-of-type(odd) td {
 To improve the {{Glossary("accessibility", "accessibility")}} of the [previous example](#column_and_row_spanning) and to allow screen readers, for example, to speak the headers associated with each header cell, the [`headers`](#headers) attribute can be introduced along with [`id`](/en-US/docs/Web/HTML/Global_attributes/id) attributes. Because of the way the "Pronunciation" column is split into t ...
 
-The [visual result](#result_2) is unchanged from the [previous example table](#column_and_row_spanning).
+> **Note:** It's recommended to use a more unique value for the [`id`](/en-US/docs/Web/HTML/Global_attributes/id) attribute. The `id` values are kept simple in this example to focus on the concept of the [`headers`](#headers) attribute.
```
as suggested previously:
```suggestion
> **Note:** It's recommended to use more descriptive and useful values for the [`id`](/en-US/docs/Web/HTML/Global_attributes/id) attribute. Each `id` in a document must be unique to that document. In this example, the `id` values are single characters to maintain focus on the concept of the [`headers`](#headers) attribute.
```

glmvc commented on files/en-us/web/html/element/thead/index.md:
```diff
@@ -197,16 +179,26 @@ In order to allocate the header cells to the correct columns and rows, the [`col
 
+#### Cell spanning
+
+In order to allocate the header cells to the correct columns and rows, the [`colspan`](/en-US/docs/Web/HTML/Element/th#colspan) and [`rowspan`](/en-US/docs/Web/HTML/Element/th#rowspan) attributes are used on the {{HTMLElement("th")}} elements. The values set in these attributes specify how many cells each header cell ({{HTMLElement("th")}} element) is to span. Due to the way these attributes are  ...
```
"... span one row and one **column** as..." - use singular
```suggestion
In order to associate and line up the header cells with the correct columns and rows, the [`colspan`](/en-US/docs/Web/HTML/Element/th#colspan) and [`rowspan`](/en-US/docs/Web/HTML/Element/th#rowspan) attributes are used on the {{HTMLElement("th")}} elements. The values set in these attributes specify how many cells each header cell ({{HTMLElement("th")}}) spans. Due to the way these attributes are set, the two second-row header cells are lined up with the columns they head. These each span one row and one column as the default values for the [`colspan`](/en-US/docs/Web/HTML/Element/th#colspan) and [`rowspan`](/en-US/docs/Web/HTML/Element/th#rowspan) attributes are both `1`.

The column and row spanning demonstrated by this example are illustrated in the following figure:
```

glmvc commented on files/en-us/web/html/element/tr/index.md:
```diff
@@ -102,6 +81,10 @@ The CSS {{cssxref(":nth-of-type")}} pseudo-class is used to select every `odd` r
 
+#### CSS
+
+The CSS {{cssxref(":nth-of-type")}} pseudo-class is used to select every `odd` row and set the {{cssxref("background-color")}} of those rows to a slightly darker tone, creating a so-called "zebra stripe" effect. This alternating background makes the rows of data in the table easier to parse and read—imagine having many rows and columns and trying to find some data in a particular row. In addition ...
```
I think it's better. But I'm not sure about the last part "..., which have a transparent...". What do you mean by that?
Originally, I just wanted to show the use of the tr selector in combination with a cell, in this example the `th` element with the scope attribute.

glmvc committed:
Apply suggestions from code review 2.0

grammar and sentence structure fixes, added infos, tweaks and final decisions

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Apply linter suggestions from reviewdog

formatting

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc committed:
Apply suggestions from code review

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

glmvc committed:
Apply linter suggestion from reviewdog

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

glmvc commented:
@estelle Thank you, most of the things were fine and a further significant improvement :)
I will go through all the pages again and if necessary adapt them to the latest suggestions and adjustments.

But I would like to check one thing with you first:
On the `th` page, you adjusted the sentence structure a bit for the attributes. Originally, I had made sure that the attribute descriptions for all table-related elements start with a verb such as " Specifies" or " Defines"... How should this be done "correctly"? Or should I adapt this for the `th` page to match the other element pages again?

estelle commented:
> On the `th` page, you adjusted the sentence structure a bit for the attributes. Originally, I had made sure that the attribute descriptions for all table-related elements start with a verb such as " Specifies" or " Defines"... How should this be done "correctly"? Or should I adapt this for the `th` page to match the other element pages again?

We don't need the word "contains" because anything that has a value contains the value. I was trying to make it less wordy. You don't have to update everything. Similarly, "specify" isn't really needed either. You don't have to remove all of them, but an action verb isn't required. 

example: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meter - 
high:
"The lower numeric bound of the high end of the measured range." could have been written as "specifies the lower numeric bound of the high end of the measured range." but it's fine as is.
Optimal: 
"This attribute indicates the optimal numeric value." could have also been written as "The optimal numeric value.", but we don't need to change it.

estelle commented on files/en-us/web/html/element/th/index.md:
```diff
@@ -16,16 +16,16 @@ The **`<th>`** [HTML](/en-US/docs/Web/HTML) element defines a cell as the header
 - `scope`
 
-  - : This {{Glossary("enumerated", "enumerated")}} attribute defines the cells that the header (defined in the `<th>`) element relates to. Possible values are:
+  - : Defines the cells that the header (defined in the `<th>`) element relates to. Possible {{Glossary("enumerated", "enumerated")}} values are:
```
the original was correct; deleted my suggested error. Thank you. Sorry about that.

glmvc committed:
Apply suggestion for colgroup element

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

estelle reviewed (approved):
Wow. A huge thanks!  And, yay! 🎉

glmvc commented on files/en-us/web/html/element/tr/index.md:
```diff
@@ -21,61 +21,40 @@ The following attributes are deprecated and should not be used. They are documen
-  - : This attribute specifies the alignment of the content to a character in row cells. Typical values for this include a period (`.`) when attempting to align numbers or monetary values. If [`align`](#align) is not set to `char`, this attribute is ignored.
-
-    > **Note:** Do not use this attribute as it is deprecated (and not supported) in the latest standard: use the CSS {{cssxref("text-align")}} property instead.
+  - : Specifies the alignment of the content to a character of each row cell. Typical values for this include a period (`.`) when attempting to align numbers or monetary values. If [`align`](#align) is not set to `char`, this attribute is ignored.
```
I am not able to get the align & char attributes to work on specific table elements in modern browsers (Firefox and Chrome). Also, I can't find any good information on how exactly this was adopted by browsers in early 2000 and how it's handled in today's browsers (if it's handled at all)... Maybe you know more or have a good source where something like this can be looked up?

This is what I found in the [old spec](https://www.w3.org/TR/html4/struct/tables.html#adef-align-TD):
> The alignment of cell contents can be specified on a cell by cell basis, or inherited from enclosing elements, such as the row, column or the table itself.

>
>The order of precedence (from highest to lowest) for the attributes [align](https://www.w3.org/TR/html4/struct/tables.html#adef-align-TD), [char](https://www.w3.org/TR/html4/struct/tables.html#adef-char), and [charoff](https://www.w3.org/TR/html4/struct/tables.html#adef-charoff) is the following:
>- An alignment attribute set on an element within a cell's data (e.g., [P](https://www.w3.org/TR/html4/struct/text.html#edef-P)).
>- An alignment attribute set on a cell ([TH](https://www.w3.org/TR/html4/struct/tables.html#edef-TH) and [TD](https://www.w3.org/TR/html4/struct/tables.html#edef-TD)).
>- An alignment attribute set on a column grouping element ([COL](https://www.w3.org/TR/html4/struct/tables.html#edef-COL) and [COLGROUP](https://www.w3.org/TR/html4/struct/tables.html#edef-COLGROUP)). When a cell is part of a multi-column span, the alignment property is inherited from the cell definition at the beginning of the span.
>- An alignment attribute set on a row or row grouping element ([TR](https://www.w3.org/TR/html4/struct/tables.html#edef-TR), [THEAD](https://www.w3.org/TR/html4/struct/tables.html#edef-THEAD), [TFOOT](https://www.w3.org/TR/html4/struct/tables.html#edef-TFOOT), and [TBODY](https://www.w3.org/TR/html4/struct/tables.html#edef-TBODY)). When a cell is part of a multi-row span, the alignment property is inherited from the cell definition at the beginning of the span.
>- An alignment attribute set on the table ([TABLE](https://www.w3.org/TR/html4/struct/tables.html#edef-TABLE)).
>- The default alignment value.
>
>Furthermore, when rendering cells, horizontal alignment is determined by columns in preference to rows, while for vertical alignment, rows are given preference over columns.

>
>The default alignment for cells depends on the user agent. However, user agents should substitute the default attribute for the current directionality (i.e., not just "left" in all cases).

estelle committed:
Merge branch 'main' into fixing-table-content

estelle commented on files/en-us/web/html/element/tr/index.md:
```diff
@@ -11,506 +11,338 @@ The **`<tr>`** [HTML](/en-US/docs/Web/HTML) element defines a row of cells in a
 
-The HTML is similar to the previous example's, except for the addition of the new column in each data row, and the changes to the header. Those changes make the HTML look like this:
+An additional table row (`<tr>`) is added as the first row of the table with column header cells ({{HTMLElement("th")}}) providing a header for each column. We put this row in a {{HTMLElement("thead")}} grouping element to indicate this is the header of the table. The [`scope`](/en-US/docs/Web/HTML/Element/th#scope) attribute is added to each header cell (`<th>`) within this head row to explicitl ...
+
```
```suggestion
```

estelle committed:
Update files/en-us/web/html/element/tr/index.md

glmvc commented:
Thanks for your suggestions, fixes and help! 🚀
